pkg/server/share_manager.go

```diff
@@ -47,7 +69,31 @@ func NewShareManager(logger logrus.FieldLogger, volume volume.Volume) (*ShareMan
 	}
 	m.context, m.shutdown = context.WithCancel(context.Background())
 
-	nfsServer, err := nfs.NewServer(logger, configPath, types.ExportPath, volume.Name)
+	enableFastFailover := m.getEnvAsBool(EnvKeyFastFailover, false)
+	leaseLifetime := m.getEnvAsInt(EnvKeyLeaseLifetime, defaultLeaseLifetime)
+	gracePeriod := m.getEnvAsInt(EnvKeyGracePeriod, defaultGracePeriod)
+
+	m.enableFastFailover = enableFastFailover
+	if m.enableFastFailover {
+		kubeclientset, err := m.NewKubeClient()
+		if err != nil {
+			m.logger.WithError(err).Warn("Failed to make lease client for fast failover")
+			return nil, err
+		}
+
+		// Use the clientset to get the node name of the share-manager pod
+		// and store for use as lease holder.
+		podName := shareManagerPrefix + m.volume.Name
+		pod, err := kubeclientset.CoreV1().Pods(namespace).Get(m.context, podName, metav1.GetOptions{})
```

Comment:
NIT: Since we are essentially just trying to get the name of the node we are currently running on, I might have preferred the use of the downward API here. However, we definitely need to have a Kubernetes client for the rest of this feature to work, so the downward API wouldn't be a particularly good optimization (as opposed to a different situation in which we didn't otherwise need a client at all.) At this point in the process, we can keep the logic as is.